huaweicloud/services/cbr/resource_huaweicloud_cbr_backup_share.go

```diff
@@ -0,0 +1,261 @@
+						},
+						"image_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The ID of the image registered with the shared backup copy.",
+						},
+						"vault_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The ID of the vault where the shared backup is stored.",
+						},
+					},
+				},
+				Description: "The list of shared members configuration.",
+			},
+		},
+	}
+}
+
+// Collect destination project IDs of all shared members.
+func buildBackupSharedMembersInfo(memberList *schema.Set) []string {
+	if memberList.Len() < 1 {
+		return nil
+	}
+	result := make([]string, memberList.Len())
+	for i, val := range memberList.List() {
+		member := val.(map[string]interface{})
+		result[i] = member["dest_project_id"].(string)
+	}
+	return result
+}
+
+func resourceBackupShareCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	client, err := cfg.CbrV3Client(cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating CBR v3 client: %s", err)
+	}
+
+	var (
+		backupId      = d.Get("backup_id").(string)
+		sharedMembers = d.Get("members").(*schema.Set)
+		opts          = members.CreateOpts{
+			BackupId: backupId,
+			Members:  buildBackupSharedMembersInfo(sharedMembers),
+		}
+	)
+
+	_, err = members.Create(client, opts)
+	if err != nil {
+		return diag.Errorf("failed while adding shared member to specified backup (%s): %s", backupId, err)
+	}
+	// Use the backup ID as the ID of this resource.
+	// Note: One backup can only manage one resource.
+	d.SetId(backupId)
+
+	return resourceBackupShareRead(ctx, d, meta)
+}
+
+func flattenBackupSharedMembersInfo(memberList []members.Member) []map[string]interface{} {
+	if len(memberList) < 1 {
+		return nil
+	}
+	result := make([]map[string]interface{}, len(memberList))
+	for i, member := range memberList {
+		result[i] = map[string]interface{}{
+			"dest_project_id": member.DestProjectId,
+			"id":              member.ID,
+			"status":          member.Status,
+			"created_at":      member.CreatedAt,
+			"updated_at":      member.UpdatedAt,
+			"image_id":        member.ImageId,
+			"vault_id":        member.VaultId,
+		}
+	}
+	return result
+}
+
+func resourceBackupShareRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.CbrV3Client(region)
+	if err != nil {
+		return diag.Errorf("error creating CBR v3 client: %s", err)
+	}
+
+	var (
+		backupId = d.Id()
+		opts     = members.ListOpts{
+			BackupId: backupId,
+		}
+	)
+	// Query the shared member list under the specified backup.
+	memberList, err := members.List(client, opts)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "CBR backup share")
+	}
+	log.Printf("[Lance] The API response of the member Create method is: %#v", memberList)
```

Comment:
please remove the log message or use DEBUG level

Reply:
fixed.